API/Repositories/Implementations/PlayerRepository.cs

```diff
@@ -60,7 +60,24 @@ public async Task<IEnumerable<Player>> GetAsync(bool eagerLoad)
         return await _context.Players.AsNoTracking().ToListAsync();
     }
 
-    public async Task<Player?> GetAsync(string username) => await Search(username).FirstOrDefaultAsync();
+    public async Task<IEnumerable<Player>> SearchAsync(string username) => await SearchQuery(username, true).ToListAsync();
+
+    public async Task<Player> GetAsync(string username)
+    {
+        List<Player> players = await SearchQuery(username, false).ToListAsync();
+
+        if (players.Count > 1)
+        {
+            throw new Exception("More than one player was found.");
+        }
+
+        if (players.Count == 0)
+        {
+            throw new Exception("No players were found.");
+        }
+
+        return players.First();
+    }
```

Comment:
On second thought, I think this should return a nullable Player.

If the players.Count > 1, you can still throw the exception, but if count is 0, return null. Otherwise, return players.First() still. This is because throwing exceptions for not found is an awful practice.